### ISSUE TITLE
Align on the usage of the relationships in the links section

### ISSUE THREAD

**Issue by JanssenBrm** [ESA-APEx/apex_algorithms#67]
We should clearly define and align the relationship type in the links section used for openEO and OGC Processes. This would help streamline the visualization of the record in the Algorithm Catalogue.

Currently, we are using the following. Some of them have a mixed usage:

| rel | type | description |
| --- | --- | --- |
| openeo-process | application/json | URL to the UDP |
| service | application/json | Link to the openEO backend or to the OGC API Processes endpoint |
| example | application/json | Example output in STAC format or used as examples for input parameters  | 
| example | applicaiton/netcdf | Example output in NetCDF |
| application | application/cwl+yaml | URL to CWL file |
| order | text/html | Link to the HTML page to order the service (custom/NoR)
| license | application/pdf | Link to the license file |
| preview | image/png | Example screenshot of the output | 
| documentation | text/html | Link to the technical documentation |

A suggestion (open to feedback):

| rel | type | description |
| --- | --- | --- |
| application | application/vnd.openeo+json;type=process | URL to the UDP |
| application | application/cwl+yaml | URL to CWL file |
| service | application/json | Link to the openEO backend or to the OGC API Processes endpoint |
| example-output | application/json | Example output in STAC format | 
| example-output | application/netcdf | Example output in NetCDF |
| example-input| application/json | Example of input parameters |
| preview | image/png | Preview images to use in the visualization | 
| order | text/html | Link to the HTML page to order the service (custom/NoR)
| license | application/pdf | Link to the license file |
| about | text/html | Link to the technical documentation |
| webapp | text/html | Link to the web application for executing the service |

> **IMPORTANT**
> The type column only contains examples based on the current records and may contain additional mime types. This ticket mainly focuses on the relationships `rel`.

**Comment by JanssenBrm** [comment]
Tagging @jdries @soxofaan @p3dr0 @HerveCaumont  to get your feedback on this.

**Comment by JanssenBrm** [comment]
Some considerations to take into account:

- Size of the output examples
- The mime type should be more specific. JSON can represent a STAC item or a plain JSON file
- The expectations should be matched as the service might not generate STAC item.

**Comment by JanssenBrm** [comment]
I've made some final updates to the relationship types. Unless there are any big objections, I suggest implementing them to close this topic.

**Comment by soxofaan** [comment]
about this part of the proposal:

> rel=application  type=application/json  URL to the UDP

this obfuscates the fact that it's a link to an openEO UDP (originally/currently it's "rel=openeo-process").  Is this intentional, e.g. to make it generic/flexible?

**Comment by JanssenBrm** [comment]
Thank you for the feedback @soxofaan!

> this obfuscates the fact that it's a link to an openEO UDP (originally/currently it's "rel=openeo-process"). Is this intentional, e.g. to make it generic/flexible?

Correct, we want to align the relationship types to make them more generic and flexible. The details should be more reflected in the MIME type. However, you have a valid comment that json might be a bit too generic. Would it be an option to use `application/udp+json` (similar to `application/cwl+yaml` for OGC API Processes)?

**Comment by soxofaan** [comment]
related discussion:
- https://github.com/Open-EO/openeo-api/issues/532

So with current state of that ticket I think we could go for `application/vnd.openeo.process+json` to link to a "UDP" (or more generically: a [remote openEO  process definition"](https://github.com/Open-EO/openeo-api/tree/draft/extensions/remote-process-definition) )

**Comment by m-mohr** [comment]
- I find that output and input are a bit weird. They feel to generic for its purpose as examples.
- Why is license required to be PDF? text/plain or text/html could be potential alternatives that are commonly used.
- docs should potentially be about, which I've seen used for documentation and is also used in openEO: https://api.openeo.org/#section/API-Principles/Web-Linking

**Comment by JanssenBrm** [comment]
> I find that output and input are a bit weird. They feel to generic for its purpose as examples.

This was also something we were talking about. Initially this was set to `output-example` but it contained two words. Do you have any suggestions from your experience?

> Why is license required to be PDF? text/plain or text/html could be potential alternatives that are commonly used. Generally, across all rel types the media types seem pretty restrictive?!

The types are just examples based on the current content of the catalogue. The goal is to restrict the `rel` values but allow more flexibility on the `type`. 

> docs should potentially be about, which I've seen used for documentation and is also used in openEO: https://api.openeo.org/#section/API-Principles/Web-Linking

Good suggestion

**Comment by m-mohr** [comment]
Two words are fine, even offical IANA rel types have multiple words, e.g. `terms-of-service`. See https://www.iana.org/assignments/link-relations/link-relations.xhtml OGC even uses URIs (sic!) such as `http://www.opengis.net/def/rel/ogc/1.0/queryables` as relation types.

**Comment by p3dr0** [comment]
regarding the mime types as @JanssenBrm mentioned we were considering this "as examples" for all relations but this reminds me that we forgot to consider their cardinality
should we explicitly allow (and also support this on the APEx GUI) to have multiple entries with the same relation but different mime-type ?

**Comment by soxofaan** [comment]
> Please consider https://github.com/Open-EO/openeo-api/issues/532#issuecomment-2609581630 before moving forward.

I understand your concern about the risks of confusion in existing clients when deviating from `application/json`  for JSON documents, but in this case we are developing new documents and new tools around that, so why not get it right from the start?

**Comment by m-mohr** [comment]
I don't know the exact context and the plans (i.e. didn't know about "developing new documents and new tools"), I just want to ensure we don't face issues in the core openEO tooling. :-)

**Comment by JanssenBrm** [comment]
I think your point is valid @m-mohr to take the openEO tooling into account. However, we are currently not interacting with any openEO tools (AFAIK), so it could make sense to update our catalogue for now with the suggestion and come back to this if openEO tooling would be required or the final decision on the type is in place.

**Comment by m-mohr** [comment]
Fair enough. I'd propose to use the new proposal in https://github.com/Open-EO/openeo-api/issues/532#issue-2227418987 as this is much easier to register with IANA.
